### PR TITLE
001/frontend prettylist

### PR DESCRIPTION
**Pull request by Valkrys**

### CODE REVIEW

**Comment by reece-bennett** [comment]
👍

File: app/src/app/data.service.ts [hunk 1]
```diff
@@ -18,15 +23,23 @@ export class DataService {
...
+  public capability: Capability;
+  public band: Band;
+  public family: Family;
+  public roles: Roles;
   public carouselRole: CarouselRole[] = [];
+
   constructor(private http: HttpClient) {
-    this.getUser();
-    this.getRoleSpecification();
+    this.getBand();
```
Comment (micalexis) [inline app/src/app/data.service.ts]: these should be call at the point at which they are required

File: app/src/app/data.service.ts [hunk 2]
```diff
@@ -18,15 +23,23 @@ export class DataService {
...
     this.getKeyDetails();
+    this.getRoles();
+    this.getRoleSpecification();
     this.getTimelineRoles();
-    this.getCarouselRoleDetails();
-   }
+    this.getUser();
+  }
 
    public getUser(): void {
```
Comment (micalexis) [inline app/src/app/data.service.ts]: your getters should return an `Observable`, getters dont return void, the clue is in the name

File: app/src/app/roles-page/capability-filter.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Capability } from '../capability';
+
+@Pipe({
+  name: 'capabilityFilter'
+})
+export class CapabilityFilterPipe implements PipeTransform {
+
+  transform(capabilities: Capability[], id: string): Capability[] {
+    if (!capabilities) return [];
```
Comment (micalexis) [inline app/src/app/roles-page/capability-filter.pipe.ts]: wrap body with brackets

File: app/src/app/roles-page/roles-filter.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { RolesFilterPipe } from './roles-filter.pipe';
+
+describe('RolesFilterPipe', () => {
```
Comment (micalexis) [inline app/src/app/roles-page/roles-filter.pipe.spec.ts]: add some more tests to test filtering

File: app/src/app/roles-page/roles-filter.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@Pipe({
+  name: 'rolesFilter'
+})
+export class RolesFilterPipe implements PipeTransform {
+
+  transform(roles: any[], searchText: string, selectedBand: string, selectedCapability: string,selectedFamily: string): any {
+    if (!roles) return [];
+    if (!searchText && !selectedBand && !selectedCapability && !selectedFamily) return roles;
+
+    // console.log(`Filtering '${searchText}' Fam:${selectedFamily} Cap:${selectedCapability} Ban:${selectedBand}`);
```
Comment (micalexis) [inline app/src/app/roles-page/roles-filter.pipe.ts]: use loggers instead

File: app/src/app/data.service.ts (hunk 2 above)
Comment (reece-bennett) [reply app/src/app/data.service.ts]: This will be changed in #36

File: app/src/app/data.service.ts (hunk 1 above)
Comment (reece-bennett) [reply app/src/app/data.service.ts]: This will be changed in #36

File: app/src/app/roles-page/roles-filter.pipe.spec.ts (hunk 4 above)
Comment (Valkrys) [reply app/src/app/roles-page/roles-filter.pipe.spec.ts]: done

**Pull request merged**